docs/pages/inboxes/sync-preferences.mdx

```diff
@@ -0,0 +1,128 @@
+
+   <div>
+      <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/pg-1.png" width="300px" />
+   </div>
+
+3. Inbox ID Alix then logs in to an Installation B of App B on their phone. The next time Installation A runs `preferences.sync`, `syncAll`, or `streamAll`, it updates the preferences group as follows:
+    
+   <div>
+      <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/pg-2.png" width="300px" />
+   </div>
+    
+4. Then let's say Inbox ID Alix logs in to an Installation C of App A on their tablet. The next time Installation A or B runs `preferences.sync`, `syncAll`, or `streamAll`, it updates the preferences group as follows:
+    
+   <div>
+      <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/pg-3.png" width="300px" />
+   </div>
+    
+## Preferences sync worker
+
+Now, let's describe how the preferences sync worker helps keep user consent preferences in sync across existing app installations.
+
+1. Let's say Inbox ID Alix uses Installation A to block Inbox ID Bo.
+
+   <div>
+      <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/a-block-bo.png" width="200px" />
+   </div>    
+    
+2. This sends a message to the preferences group in the group message database. This is not an actual chat message, but a serialized proto message that is not shown to app users.
+    
+   <div>
+      <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/gm-pg.png" width="300px" />
+   </div>    
+    
+3. When Installation B calls `preferences.sync`, `syncAll`, or `streamAll`, it gets the message from the preferences group. The sync worker listens for these preferences group messages and processes the message to block Inbox ID Bo in Installation B.
+    
+   <div>
+      <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/b-block-bo.png" width="200px" />
+   </div> 
+    
+4. Likewise, when Installation C calls `preferences.sync`, `syncAll`, or `streamAll`, it gets the message from the preferences group, and the sync worker ensures Inbox ID Bo is blocked there as well.
+    
+   <div>
+      <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/c-block-bo.png" width="200px" />
+   </div>   
+
+Preferences sync handles syncing HMAC keys in the same way.
+
+For example, it will handle syncing an HMAC key shared by Installation B to other installations. When Installation A and C call `preferences.sync`, `syncAll`, or `streamAll`, the sync worker gets Installation B's HMAC key message from the preferences group and ensures that Installation A and C get the HMAC key for Installation B.
+
+## Sync preferences
+
+Use this call to sync preferences (consent preferences and HMAC keys) across app installations in a preferences group. 
```

Comment:
```suggestion
Use this call to sync preferences (consent preferences and HMAC keys) across app installations in a preferences group. This will also sync welcomes to ensure that you have all potential new installations before syncing.
```